voter-dapp/src/ActiveRequests.js

```diff
@@ -7,14 +7,34 @@ import TableBody from "@material-ui/core/TableBody";
 import TableCell from "@material-ui/core/TableCell";
 import TableHead from "@material-ui/core/TableHead";
 import TableRow from "@material-ui/core/TableRow";
+import TextField from "@material-ui/core/TextField";
 import Typography from "@material-ui/core/Typography";
 
 import { formatDate, formatWei } from "./common/FormattingUtils.js";
 import { VotePhasesEnum } from "./common/Enums.js";
-import { decryptMessage, deriveKeyPairFromSignatureMetamask } from "./common/Crypto.js";
+import { decryptMessage, deriveKeyPairFromSignatureMetamask, encryptMessage } from "./common/Crypto.js";
 const { getKeyGenMessage } = require("./common/EncryptionHelper.js");
 
+const editStateReducer = (state, action) => {
```

Comment:
I think this simplifies things a lot - nice!

Reply:
I didn't really the point of reducers at first, but with this example in hand, I'm beginning to understand where they could be useful.